src/librustdoc/html/static/js/search.js

```diff
@@ -590,10 +670,10 @@ function initSearch(rawSearchIndex) {
      * The supported syntax by this parser is as follow:
      *
      * ident = *(ALPHA / DIGIT / "_") [!]
-     * path = ident *(DOUBLE-COLON ident)
+     * path = ident *(DOUBLE-COLON/{WS}- ident)
      * arg = path [generics]
      * arg-without-generic = path
-     * type-sep = COMMA/WS *(COMMA/WS)
+     * type-sep = COMMA *(COMMA)
```

Comment:
The most recent talk about this was in https://github.com/rust-lang/rust/pull/90630. Otherwise it goes way before 1.0 as far as I see. Quite the old choice.

> Whether we wind up making this change or not, the final decision should be added to https://github.com/rust-lang/rust/blob/master/src/doc/rustdoc/src/how-to-read-rustdoc.md#the-search-interface so that users know what's going on.

Agreed. I'll do it if we agree that we want this change (and that we want it this way).

Reply:
Didn't find where the whitespace was introduced so thanks for that!

So I guess with this, we're putting back the original meaning.